### PR TITLE
Update Auto-sklearn benchmark

### PR DESCRIPTION
**Pull request by mfeurer**

### CODE REVIEW

**Comment by KEggensperger** [comment]
@mfeurer This is not yet ready, correct? Could you ping me once you think this is ready to be merged?

File: hpolib/benchmarks/ml/autosklearn_benchmark.py [hunk 1]
```diff
@@ -85,10 +111,10 @@ def _get_data_manager(self, task_id):
             # TODO in the future, the XYDataManager should have this as it's own
```
Comment (KEggensperger) [inline hpolib/benchmarks/ml/autosklearn_benchmark.py]: Will this TODO ever be solved?

File: optional-requirements.txt [hunk 2]
```diff
@@ -9,8 +9,8 @@ scikit-learn>=0.19.0,<0.20.0
 openml
 
 # for benchmarks using auto-sklearn
-auto-sklearn==0.3.0
 pynisher
```
Comment (KEggensperger) [inline optional-requirements.txt]: auto-sklearn requires pynisher==0.5.0, shouldn't this be also stated here?

File: requirements.txt [hunk 3]
```diff
@@ -6,5 +6,5 @@ scikit-learn>=0.19.0,<0.20.0
 liac-arff>=2.1.1
 ConfigSpace>=0.4.0
 openml
-pynisher
```
Comment (KEggensperger) [inline requirements.txt:9]: Not entirely sure, why the pynisher is in here, I guess only auto-sklearn needs it?

File: hpolib/benchmarks/ml/autosklearn_benchmark.py (hunk 1 above)
Comment (mfeurer) [reply hpolib/benchmarks/ml/autosklearn_benchmark.py]: It's actually solved, I just didn't see the todo.

File: optional-requirements.txt (hunk 2 above)
Comment (mfeurer) [reply optional-requirements.txt]: Good point, will do.

File: requirements.txt (hunk 3 above)
Comment (mfeurer) [reply requirements.txt:9]: Good point. We only need it for testing and it is specified in the test requirements. I will remove it.

File: hpolib/benchmarks/ml/autosklearn_benchmark.py [hunk 4]
```diff
@@ -30,22 +30,48 @@ class AutoSklearnBenchmark(AbstractBenchmark):
     auto-sklearn benchmarks implement Section 6 of the paper 'Efficient and
     Robust Automated Machine Learning' by Feurer et al., published in
     Proceedings of NIPS 2015.
+
```
Comment (KEggensperger) [inline hpolib/benchmarks/ml/autosklearn_benchmark.py:33]: Shouldn'T this comment be inside __init__?

File: hpolib/benchmarks/ml/sklearn_mlp_benchmark.py [hunk 5]
```diff
@@ -175,9 +175,3 @@ def get_meta_information():
                     "publisher={IEEE}"]
         d["references"].append(data_ref)
         return d
-
```
Comment (KEggensperger) [inline hpolib/benchmarks/ml/sklearn_mlp_benchmark.py:178]: Is removing the benchmark really part of this PR?

**Review by KEggensperger — COMMENTED** [review]
I think this is ready to be merged, but could you please check the few comments.

File: hpolib/benchmarks/ml/autosklearn_benchmark.py (hunk 4 above)
Comment (mfeurer) [reply hpolib/benchmarks/ml/autosklearn_benchmark.py:33]: I'm always unsure where such comments should be, but basically, this is about the class itself, not about the init method.

**Comment by KEggensperger** [comment]
@mfeurer Can you have a look at the error message on Travis? If this is solved, we can finally merge ...

```
ERROR: test_load_openml (test_openml_data_manager.TestDataManager)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/automl/HPOlib2/tests/test_openml_data_manager.py", line 10, in test_load_openml
    X_train, y_train, X_val, y_val, X_test, y_test = dm.load()
  File "/home/travis/build/automl/HPOlib2/hpolib/util/openml_data_manager.py", line 94, in load
    self.task_id)
  File "/home/travis/build/automl/HPOlib2/hpolib/util/openml_data_manager.py", line 14, in _load_data
    task = openml.tasks.get_task(task_id)
  File "/home/travis/miniconda/envs/testenv/lib/python3.5/site-packages/openml/tasks/functions.py", line 310, in get_task
    raise e
  File "/home/travis/miniconda/envs/testenv/lib/python3.5/site-packages/openml/tasks/functions.py", line 301, in get_task
    task.download_split()
  File "/home/travis/miniconda/envs/testenv/lib/python3.5/site-packages/openml/tasks/task.py", line 116, in download_split
    split = OpenMLSplit._from_arff_file(cached_split_file)
  File "/home/travis/miniconda/envs/testenv/lib/python3.5/site-packages/openml/tasks/split.py", line 90, in _from_arff_file
    type_idx = meta._attrnames.index('type')
AttributeError: 'MetaData' object has no attribute '_attrnames'
```

**Comment by mfeurer** [comment]
That's actually an issue with OpenML at the moment, not sure what happened there. I can check this after Friday.

**Pull request merged**